iceoryx_examples/automotive_soa/include/owl/kom/field_subscriber.inl

```diff
@@ -0,0 +1,182 @@
+               .and_then([&](const auto& sample) {
+                   callable(sample.get());
+                   numberOfSamples++;
+               })
+               .or_else([](auto& result) {
+                   if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                   {
+                       std::cerr << "Error receiving chunk!" << std::endl;
+                   }
+               })
+           && numberOfSamples < maxNumberOfSamples)
+    {
+    }
+    return numberOfSamples;
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Get()
+{
+    // If we call Get() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    //! [FieldSubscriber get]
+    bool requestSuccessfullySent{false};
+    m_client.loan()
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request.send().and_then([&]() { requestSuccessfullySent = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySent = false;
+        });
+
+    if (!requestSuccessfullySent)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+    //! [FieldSubscriber get]
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Set(const FieldType& value)
+{
+    // If we call Set() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    bool requestSuccessfullySent{false};
+    m_client
+        .loan()
+        //! [FieldSubscriber set]
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request->emplace(value);
+            //! [FieldSubscriber set]
+            request.send().and_then([&]() { requestSuccessfullySent = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySent = false;
+        });
+
+    if (!requestSuccessfullySent)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::receiveResponse()
+{
+    Promise<FieldType> promise;
+    auto future = promise.get_future();
+    m_threadsRunning++;
+    // Typically you would e.g. use a worker pool here, for simplicity we use a plain thread
+    std::thread(
+        [&](Promise<FieldType>&& promise) {
+            // Avoid race if FieldSubscriber d'tor is called while this thread is still running
```

Comment:
This was solved by `m_threadsRunning` therefore we could remove this mutex unless something else is guarded with `m_mutex`. Maybe we can rename the mutex a bit so that we reviewers know what the mutex should protect.